lib/rules/no-deprecated.js

```diff
@@ -73,6 +76,22 @@ module.exports = {
       deprecated[`${pragma}.PropTypes`] = ['15.5.0', 'the npm module prop-types'];
       // 15.6.0
       deprecated[`${pragma}.DOM`] = ['15.6.0', 'the npm module react-dom-factories'];
+      // 16.3.0
+      deprecated.componentWillMount = [
+        '16.3.0',
+        'constructor',
```

Comment:
hmm, i'm actually not comfortable recommending the constructor here - side-effecting things should never go in a constructor, and there's still some use cases for `componentWillMount` that aren't covered by "constructor" or "getDerivedStateFromProps".

I wonder if maybe we should also add an "ignore" array of method names; ie so I can configure the rule to ignore `componentWillMount` but warn on the rest.

Reply:
How about recommending `componentDidMount()` (or [`UNSAFE_componentWillMount()`](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount) as a temporary measure)?

Reply:
I guess the unsafe one; but it'd still probably be a good idea to add an ignore list.

Reply:
Agree, in terms of consistency it would be better to deprecated them in favor of `UNSAFE_` methods, it should allow to avoid confusion. @ljharb, does it sound reasonable to leave warnings with `UNSAFE_` methods as new methods for now and deprecate them in 17.0.0?

Reply:
Regarding the ignore list for that rule, I vote for adding a dedicated issue for further discussion. I could take a look at it.

Reply:
A separate issue for the "UNSAFE_" methods would be ideal.